ui/imports/shared/views/chat/CompactMessageView.qml

```diff
@@ -764,11 +764,14 @@ Item {
                 onAddEmojiClicked: {
                     if(root.isChatBlocked)
                         return
+
+                    // First set parent, X & Y positions for the messageContextMenu
+                    root.messageContextMenu.parent = emojiRect
+                    root.messageContextMenu.setXPosition = function() { return (root.messageContextMenu.parent.x + root.messageContextMenu.parent.width + 4) }
+                    root.messageContextMenu.setYPosition = function() { return  (-root.messageContextMenu.height - 4) }
+
+                    // Second, add emoji that also triggers setXYPosition methods / open popup:
```

Comment:
While this fixes the problem for sure, wouldn't it be better to replace all that imperative JS code with simple property bindings? After all, `emojiReactionLoader` already has anchors

Reply:
Well, from my point of view, what we should do here is redesign the `MessageContextMenuView` since it does a looot of things / has not-easy-to-follow variants and responsabilities are not clear.  And same comment applies to the @MishkaRogachev proposal. 

WDYT if we fix that specific issue and create a separate task to refactor that component properly?

Reply:
yeah, go for it

Reply:
@caybro @MishkaRogachev, here the issue created. Please add whatever specific information you think will be interesting: https://github.com/status-im/status-desktop/issues/6423